gerousiabot/discord_bot.py

```diff
@@ -0,0 +1,92 @@
+from typing import List
+
+from discord import Intents, Member, VoiceChannel, Client
+
+from gerousiabot import utils
+
+logger = utils.setup_logger()
+
+
+class GerousiaBot(Client):
+
+    def __init__(self, bot_token: str, **options):
+        intents = get_intents_needed_to_check_online_members()
+        logger.debug('Creating bot with with intents {}'.format(intents))
+
+        super().__init__(intents=intents, **options)
+        logger.debug('Creating bot with with valid token {}'.format(bot_token is not None))
+        self.bot_token = bot_token
```

Comment:
Is it possible to set the token as part of the `super().__init__` instead of a mutating afterthought?

Reply:
Quick noob question, you mean super().__init__(intents=intents, bot_token=bot_token **options) ? 
One less line?

Reply:
@KSBolton exactly what I meant. What we have functionally works, but I lean towards "pure functional" style of programming nowadays. No object mutations, pure functions and methods, etc. 

See points 1, 3, and 4
https://hackernoon.com/9-functional-programming-concepts-everyone-should-know-uy503u21